src/GovUk.Education.ExploreEducationStatistics.Content.Services/DataSetService.cs

```diff
@@ -34,34 +36,42 @@ public DataSetService(ContentDbContext contentDbContext)
         Guid? themeId = null,
         Guid? publicationId = null,
         Guid? releaseId = null,
-        bool? latest = true,
+        bool? latestOnly = true,
         string? searchTerm = null,
         DataSetsListRequestOrderBy? orderBy = null,
         SortOrder? sort = null,
         int page = 1,
-        int pageSize = 10)
+        int pageSize = 10,
+        CancellationToken cancellationToken = default)
     {
+        // If latestOnly is not specified default it to true except when a releaseId is specified
+        latestOnly ??= !releaseId.HasValue;
```

Comment:
Nullable value types and optional arguments are two different things.

`latestOnly` is both a nullable value type and an optional argument.

If the `latestOnly` argument is present and `true` or `false` the left operand of this line already has a value so there's no overall change.

If the `latestOnly` argument is present but `null`, the left operand of this line has no value and gets assigned the right hand expression.

If the `latestOnly` argument is omitted in the method call, it will be assigned the default value which is true. This is the only time when the default value has an affect. Then, as above, the left operand of this line already has a value so there's no overall change.

The `latestOnly` argument is provided to this method call with a `null` value when it's omitted from the controller request, so this line is having an affect currently. See `DataSetsListRequest` where `LatestOnly` is an optional value type so it will be null if omitted in the request.

It's probably clearer if I get rid of all the optional arguments on `DataSetService.ListDataSets` since we always expect the controller to call the service with every argument. This is a hangover from `DataSetServiceTests` where it was easier to use optional arguments on the service to prevent every using test from declaring every argument, a lot of which would be null.

Reply:
I've made this change now in 5fc2abd4d83ca2f9c1e395991abbba5e56130664.